src/main/java/org/kohsuke/github/GHRelease.java

```diff
@@ -122,18 +126,36 @@ GHRelease wrap(GHRepository owner) {
      * http://stackoverflow.com/questions/12361090/server-name-indication-sni-on-java but involve more complicated
      * handling of the HTTP requests to github's API.
      *
+     * @param file file to upload
+     * @param contentType content type
+     * @param uploadUrlPrefix URL prefix for uploading; for example, for github.com it's https://uploads.github.com
+     * @return instance of GHAsset uploaded
+     * @since 1.75
      * @throws IOException
      */
-    public GHAsset uploadAsset(File file, String contentType) throws IOException {
+    public GHAsset uploadAsset(File file, String contentType, String uploadUrlPrefix) throws IOException {
         Requester builder = new Requester(owner.root);
 
-        String url = format("https://uploads.github.com%s/releases/%d/assets?name=%s",
-                owner.getApiTailUrl(""), getId(), file.getName());
+        String url = format("%s%s/releases/%d/assets?name=%s",
+                            uploadUrlPrefix, owner.getApiTailUrl(""), getId(), file.getName());
         return builder.contentType(contentType)
                 .with(new FileInputStream(file))
                 .to(url, GHAsset.class).wrap(this);
     }
 
+    /**
+     * upload the release to defaul upload URL on github.com
+     *
+     * @param file file to upload
+     * @param contentType content type
+     * @return instance of GHAsset uploaded
+     * @throws IOException
+     */
+    public GHAsset uploadAsset(File file, String contentType) throws IOException {
```

Comment:
ok, done and pushed. Thanks!





---- On Ter, 05 Abr 2016 16:25:58 +0200 Merkushev Kirill &lt;notifications@github.com&gt;wrote ---- 




In src/main/java/org/kohsuke/github/GHRelease.java:

&gt; return builder.contentType(contentType) &gt; .with(new FileInputStream(file)) &gt; .to(url, GHAsset.class).wrap(this); &gt; } &gt; &gt; + /** &gt; + * upload the release to defaul upload URL on github.com &gt; + * &gt; + * @param file file to upload &gt; + * @param contentType content type &gt; + * @return instance of GHAsset uploaded &gt; + * @throws IOException &gt; + */ &gt; + public GHAsset uploadAsset(File file, String contentType) throws IOException { 
yep!

 You can try defaultIfEmpty() from commons-lang (if available) instead of uploadUrlPrefix != null ? uploadUrlPrefix : DEFAULT_UPLOAD_URL;


and call return this.uploadAsset(file, contentType, DEFAULT_UPLOAD_URL);

also please reflect in javadoc, that third param may be null and default value will be used instead

—

You are receiving this because you authored the thread.

Reply to this email directly or view it on GitHub